rusk/src/lib/services/prover.rs

```diff
@@ -71,17 +71,16 @@ impl RuskProver {
         let outputs = tx.outputs();
         let proof = tx.proof();
 
-        let circuit = circuit_from_numbers(inputs.len(), outputs.len())
-            .ok_or_else(|| {
-                Status::invalid_argument(format!(
-                    "Expected: 0 < (inputs: {}) < 5, 0 ≤ (outputs: {}) < 3",
-                    inputs.len(),
-                    outputs.len()
-                ))
-            })?;
+        let circuit = circuit_from_numbers(inputs.len()).ok_or_else(|| {
+            Status::invalid_argument(format!(
```

Comment:
Is this still triggered when `outputs>2`? Cannot find the check

Reply:
You're right, the check is missing. Good catch!

Reply:
https://github.com/dusk-network/rusk/pull/694/commits/eb80fcdef99ad9478d90ab974fb5d31319335963